src/app/[locale]/(pages)/accedi/errore/__test__/errore.test.tsx

```diff
@@ -0,0 +1,117 @@
+import LoginErrorPage from '../page';
+import { pushMock } from '../../../../../../../jest.setup';
+import { renderWithProviders } from '@/app/[locale]/_utils/test-utils';
+import { ROUTES } from '@/app/[locale]/_utils/routes';
+
+// Mock the useSearchParams function
+jest.mock('next/navigation', () => ({
+  useSearchParams: jest.fn(),
+}));
+
+// Mock the behavior of useSearchParams
+const mockSearchParams = {
+  get: jest.fn(),
+};
+
+(useSearchParams as jest.Mock).mockReturnValue(mockSearchParams);
+
+describe('test suite for Accedi Errore ', () => {
+  test('should render "Generic error" when errorCode is null', async () => {
+    mockSearchParams.get.mockReturnValue(null);
+    await renderWithProviders(<LoginErrorPage />);
+    const errorSummary = screen.getByText(
+      'Si è verificato un errore durante il login, si prega di riprovare!'
+    );
+    expect(errorSummary).toBeInTheDocument();
+  });
+
+  test('should render "Credential error" for errorCode 19', async () => {
+    mockSearchParams.get.mockReturnValue('19');
+    await renderWithProviders(<LoginErrorPage />);
+    const errorSummary = screen.getByText(
+      'Hai inserito troppe volte un nome utente o una password non corretti: riprova fra qualche minuto o contatta il tuo fornitore di identità SPID per cambiare le tue credenziali.'
+    );
+    expect(errorSummary).toBeInTheDocument();
+  });
+
+  test('should render "Two-factor need" for errorCode 20', async () => {
+    mockSearchParams.get.mockReturnValue('20');
+    await renderWithProviders(<LoginErrorPage />);
+    const errorSummary = screen.getByText(
+      'Per motivi di sicurezza è richiesto un login con un secondo fattore di autenticazione: contatta il tuo fornitore di identità SPID per capire come si usa.'
+    );
+    expect(errorSummary).toBeInTheDocument();
+  });
+
+  test('should render "Session Expired" for errorCode 21', async () => {
+    mockSearchParams.get.mockReturnValue('21');
+    await renderWithProviders(<LoginErrorPage />);
+    const errorSummary = screen.getByText(
+      "È passato troppo tempo da quando hai iniziato l'accesso: per piacere riparti dall'inizio."
+    );
+    expect(errorSummary).toBeInTheDocument();
+  });
+
+  test('should render "Portal Consents" for errorCode 22', async () => {
+    mockSearchParams.get.mockReturnValue('22');
+    await renderWithProviders(<LoginErrorPage />);
+    const errorSummary = screen.getByText(
+      'Per accedere al portale, devi acconsentire all’invio di alcuni dati. Entra di nuovo e dai il tuo consenso.'
+    );
+    expect(errorSummary).toBeInTheDocument();
+  });
+
+  test('should render "Spid Revoked" for errorCode 23', async () => {
+    mockSearchParams.get.mockReturnValue('23');
+    await renderWithProviders(<LoginErrorPage />);
+    const errorSummary = screen.getByText(
+      'La tua identità SPID risulta sospesa o revocata. Per riattivare il tuo account, contatta il tuo fornitore di identità SPID.'
```

Comment:
If we already have translations for the UI component texts, are we able to use them so that if they change we don't need to modify our tests?

Reply:
@claudiodall what about this comment?

Reply:
I had missed the comment, sorry, fixed.